### PR TITLE
Add asyncapi_gencpp, log_view, rpad, rpad_ros, and param_util source to rolling

### PR DESCRIPTION
**Pull request by malban**
References https://github.com/ros2-gbp/ros2-gbp-github-org/issues/88

# Please Add This Package to be indexed in the rosdistro.

rolling

# The source is here:
  * [hatchbed_common](https://github.com/hatchbed/hatchbed_common)
  * [log_view](https://github.com/hatchbed/log_view)
  * [asyncapi_gencpp](https://github.com/hatchbed/asyncapi_gencpp) 
  * [opensw](https://github.com/hatchbed/opensw)
  * [opensw_ros](https://github.com/hatchbed/opensw_ros)

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

**Review by methylDragon — CHANGES_REQUESTED** [review]
Nosetest is failing due to ordering issues

File: rolling/distribution.yaml [hunk 1]
```diff
@@ -1919,6 +1929,16 @@ repositories:
       url: https://github.com/boschglobal/locator_ros_bridge.git
       version: main
     status: maintained
+  log_vew:
```
Comment (cottsay) [inline rolling/distribution.yaml:1932]:
```suggestion
  log_view:
```

**Comment by clalancette** [comment]
I have to agree with @cottsay that the names `param_util` and `log_view` are too generic.  For that matter, while `rpad` isn't that generic, I'd also argue that it is a bad name for it.  Maybe `slamware_driver` or something like that might be better.  At the end of the day, users are going to be doing things like installing:

`ros-rolling-rpad`, and that just doesn't seem to provide enough context about it.

For reference, here is our naming guidelines for packages: https://www.ros.org/reps/rep-0144.html .  Particularly, see https://www.ros.org/reps/rep-0144.html#global-rules

**Comment by malban** [comment]
Yeah, I understand on `param_util`.  I'll come up with something else.

`log_view` has already been released in ros1 for a while, though, and also more recently in galactic: https://index.ros.org/p/log_view/github-hatchbed-log_view/

Good point about `rpad`, but I'm not sure about using a commercial trademark like `slamware`, `slamtek`, `rplidar`, etc in the name.  I'm fine with the name being a bit cryptic to avoid potential trademark liability since the package description and readme explain the use case.   I'm also open to a name that gives more context, but it's not clear to me what that would be.  Maybe `opensw` and `opensw_driver` are marginally better?

**Comment by malban** [comment]
Here are some alternate name for `param_util`:
 - parameterize
 - rosparam_helper
 - rosparam_kit
 - rosparam_assist
 - param_kit
 - param_helper
 - param_assist
 - param_aid

Let me know if any of those work

**Comment by tfoote** [comment]
Unfortunately most of those proposed names don't really help you if you want to search for this package. One easy way to think about this is what would I find if I searched for the name? If I search for `param` `rosparam` or any of the generic modifiers like `kit` how likely is it that I'll find documentation for this package? Obviously a full string match will likely find it but others not so much. 

For the trademark you don't want to pretend to be them, but if you're writing something that works with a device you can talk about how you're compatible with that device. And if you use a good searchable term then people can find the package easily.

**Comment by malban** [comment]
> Unfortunately most of those proposed names don't really help you if you want to search for this package. One easy way to think about this is what would I find if I searched for the name? If I search for param rosparam or any of the generic modifiers like kit how likely is it that I'll find documentation for this package? Obviously a full string match will likely find it but others not so much.

Yeah, I see that.

Doing a search of [rosparam](https://index.ros.org/search/?term=rosparam) on the ROS package index turns up:
 - [rosparam_shortcuts](https://index.ros.org/p/rosparam_shortcuts/github-davetcoleman-rosparam_shortcuts/) - (1st page) a similar package to help with parameters, but different approach and set of features
 - [rosparam_handler](https://index.ros.org/p/rosparam_handler/github-cbandera-rosparam_handler/#melodic) - (2nd page) a similar package, but different approach and set of features
 - [rtt_rosparam](https://github.com/orocos/rtt_ros_integration/blob/toolchain-2.9/rtt_rosparam) - (2nd page) "Plugin for synchronizing ROS parameters with Orocos component properties."
 -   core packages: `rosparam`, `ros_comm`, `test_rosparam`

Searching for [param](https://index.ros.org/search/?term=param):
 - [launch_param_builder](https://index.ros.org/p/launch_param_builder) - (1st page) "Python library for loading parameters in launch files"
 - [choreo_param_helpers](https://index.ros.org/p/choreo_param_helpers/github-yijiangh-Choreo#kinetic) - (3rd page) "Implements routines for loading, saving, and checking for the existence of user parameters."
 - core packages:  `rcl_yaml_param_parser`, `ros2param`, 

That seems pretty reasonable to me.  I think including `rosparam` in the package name gives more context than simply `param` and would certainly be more searchable outside of ROS.  

A second word like `kit` or `helper` provides more context when paired with `rosparam`, but on it's own is very generic and would pull unrelated packages into the results even paired with `rosparam`.  I'm not sure in what context someone would search for just `helper`, `kit`, `utils`, etc.

So, is `rosparam_<some acceptable modifier>` a good approach?
- rosparam_tools
   - another generic option, LOTS of `_tools` packages and results when searching for 'tools'
   - but it sounds good to me and seems the most on point
- rosparam_toolset 
- yarpp
   - YetAnotherRosParamPackage :laughing:  
- rosparam_yarp

Open to suggestions

**Comment by malban** [comment]
> For the trademark you don't want to pretend to be them, but if you're writing something that works with a device you can talk about how you're compatible with that device. And if you use a good searchable term then people can find the package easily.

Right, I agree with that.  It shouldn't be an issue to describe that the software is compatible with the devices and that is in the [README](https://github.com/hatchbed/rpad/blob/main/README.md), package.xml, etc.  It's more of putting the trademark in the actual name of the package, which, not being a lawyer, but doing some web searching, looks like is generally not recommended.

Not sure how to add more context to the package name in this case, other than my previous suggestion of using `opensw` and `opensw_driver` which might be marginally better for context.

Open to suggestions, though.

File: rolling/distribution.yaml [hunk 2]
```diff
@@ -2683,6 +2703,16 @@ repositories:
       url: https://github.com/OUXT-Polaris/ouxt_common.git
       version: master
     status: developed
+  param_util:
+    doc:
+      type: git
+      url: https://github.com/hatchbed/param_util.git
```
Comment (ivanpauno) [inline rolling/distribution.yaml:2709]: > So, is rosparam_<some acceptable modifier> a good approach?

Yes, that or `<prefix>_rosparam` would work.
But the modifier/prefix, should be something not generic, i.e. avoid "tool" or "kit".

Of the ones you listed, `choreo_param_helpers` and `rtt_rosparam` are good examples, as it's clear to what they refer to.
Example that could work for you: `hatchbed_rosparam` (you may find something better :smile: )

**Comment by hidmic** [comment]
@malban friendly ping

**Comment by malban** [comment]
> @malban friendly ping

Thanks for the reminder!

File: rolling/distribution.yaml [hunk 3]
```diff
@@ -1919,6 +1929,16 @@ repositories:
       url: https://github.com/boschglobal/locator_ros_bridge.git
       version: main
     status: maintained
+  log_view:
+    doc:
+      type: git
+      url: https://github.com/hatchbed/log_view.git
```
[the remark this answers is not in the document]
Comment (ivanpauno) [reply rolling/distribution.yaml:1935]: that sounds fine to me.
@clalancette @tfoote is that okay?

File: rolling/distribution.yaml (hunk 2 above)
Comment (ivanpauno) [reply rolling/distribution.yaml:2709]: That sounds good to me as well

**Comment by malban** [comment]
Ok, based on comments, I've renamed:
- `param_util` -> `hatchbed_common`
- `rpad` -> `opensw`
- `rpad_ros` -> `opensw_ros`

**Review by nuclearsandwich — APPROVED** [review]
The latest proposed names look acceptable to me but I think it is worth getting a ✔️ from at least one of the original reporters as well.

**Review by cottsay — APPROVED** [review]
Thanks for iterating, @malban. I think we've arrived at something better.

**Comment by clalancette** [comment]
Generally this looks good to me now.  Thanks for the iteration.

The `log_view` one is unfortunate, since it is really generic.  But it was accepted into other ROS 2 distributions, so we can take that one on precedent.

**Review by clalancette — APPROVED** [review]

**Pull request merged**